staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -79,29 +81,58 @@ func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) (bool,
 
 // CheckClient accepts client and calculate the support per endpoint and caches it.
 // It will return at any point if error happens or one endpoint is not supported.
-func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) error {
+func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, cancel context.CancelFunc, c client, feature storage.Feature) error {
 	switch feature {
 	case storage.RequestWatchProgress:
-		return f.clientSupportsRequestWatchProgress(ctx, c)
+		return f.checkClient(ctx, cancel, c)
 	default:
 		return fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
 
 	}
 }
 
-func (f *defaultFeatureSupportChecker) clientSupportsRequestWatchProgress(ctx context.Context, c client) error {
+func (f *defaultFeatureSupportChecker) checkClient(ctx context.Context, cancel context.CancelFunc, c client) error {
+	go func(ctx context.Context) {
+		// start with 10 ms, multiply by 2 each step, 16 steps ~= 10 minutes
```

Comment:
10 min is a long time, since we are not gating the server readiness on this feature during that time requests that require that feature will fall back to a default/previous behaviour which means that we will have two categories of requests which might be hard to debug. 

Wouldn't be simpler to gate the server readiness until the check is complete ?